### PR TITLE
Fixes for 2020-02 CWG Poll 1

### PR DESCRIPTION
**Pull request by zygoloid**
Improvements inspired by a comment from @hubert-reinterpretcast on #3782.

### CODE REVIEW

File: source/basic.tex [hunk 1]
```diff
@@ -3313,28 +3313,23 @@
...
+the storage that $o_1$ occupied, and
 
-\item the new object is of the same type as the original object
+\item $o_1$ and $o_2$ are of the same type
 (ignoring the top-level cv-qualifiers), and
 
-\item the original object is neither
-a complete object that is const-qualified nor
-a subobject of such an object, and
+\item $o_1$ is not a complete const object, and
```
Comment (hubert-reinterpretcast) [inline source/basic.tex]: I believe the following is almost as brief and a more direct adaptation of the existing bullet:
```suggestion
\item the complete object of $o_1$ is not a const object, and
```

Comment (zygoloid) [reply source/basic.tex]: Yes. The question is, do we want to check for a const complete object on every level of the upward recursion, or only when we reach the complete object? The latter seems simpler to me, but I don't feel at all strongly.

Comment (hubert-reinterpretcast) [reply source/basic.tex]: My thinking on this is that it is easier to get this bullet out of the way by checking it first and then ignoring it during the recursion.

Comment (jensmaurer) [reply source/basic.tex]: Assuming "the complete object of x" is x if x is a complete object, it feels that Hubert's formulation is a little clearer.

Comment (tkoeppe) [reply source/basic.tex]: It's just a trade: with the HST wording, we have to do the "complete object of" search in every recursion round.

**Pull request merged**